### ISSUE TITLE
Duplicate emails if you set the order to cancelled or pending payment

### ISSUE THREAD

**Issue by beesmart** [woocommerce/woocommerce#27791]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
An order is created and automatically assigned to On-Hold, when the administrator changes the order status to Pending Payment and then Completed, WooCommerce sends out two total notifications. The email notifications that are sent are incorrect.

Firstly when the order is created, an email notification is sent, this is the New Order email
Secondly, when the order is completed, an email notification is sent, this is also the new order email?

It is worth noting that, this seems to only occur when the order status of 'Pending Payment' has been assigned in between order created and order completion.

**Expected behavior**
When the order is created, an email notification is sent, this is the New Order email
Secondly, when the order is completed, an email notification is sent, this is the completed order email

**Actual behavior**
An order is created and automatically assigned to On-Hold, when the administrator changes the order status to Pending Payment and then Completed, WooCommerce sends out two total notifications.

Firstly when the order is created, an email notification is sent, this is the New Order email
Secondly, when the order is completed, an email notification is sent, this is also the new order email?

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
- Create an order - Email sent here
- Order should automatically assign to Order On Hold
- Set order status to Pending Payment and update order
- Set order status to completed and update order - another email sent here, though it is the wrong email

**Screenshots**
![download (1)](https://user-images.githubusercontent.com/7336428/94338185-3f6fab80-ffe8-11ea-8048-f7fdd87f5d2a.png)

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://build14.auracreativemedia.co.uk
Site address (URL): https://build14.auracreativemedia.co.uk
WC Version: 4.4.1
REST API Version: ✔ 1.0.15
WC Blocks Version: ✔ 3.1.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.4.0
Log Directory Writable: ✔
WP Version: 5.5.1
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.39 (Unix) mod_hive/6.27 OpenSSL/1.0.1e-fips mod_fastcgi/2.4.6
PHP Version: 7.2.29
PHP Post Max Size: 256 MB
PHP Time Limit: 600
PHP Max Input Vars: 3000
cURL Version: 7.59.0
OpenSSL/1.0.2r

SUHOSIN Installed: –
MySQL Version: 5.6.36-82.1-log
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.4.1
WC Database Prefix: wpmg_
Total Database Size: 3.42MB
Database Data Size: 1.98MB
Database Index Size: 1.44MB
wpmg_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpmg_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpmg_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_actionscheduler_actions: Data: 0.03MB + Index: 0.11MB + Engine InnoDB
wpmg_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wpmg_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_options: Data: 1.03MB + Index: 0.06MB + Engine InnoDB
wpmg_postmeta: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wpmg_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpmg_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wpmg_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpmg_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wpmg_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpmg_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wpmg_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpmg_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpmg_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 6
post: 2
product: 1
revision: 5
shop_order: 16
wc_membership_plan: 2
wpsl_stores: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Code Snippets: by Code Snippets Pro – 2.14.0
SG Optimizer: by SiteGround – 5.6.5
WooCommerce Memberships: by SkyVerge – 1.19.0
WooCommerce: by Automattic – 4.4.1 (update to version 4.5.2 is available)

### Inactive Plugins (6) ###

Aura Store Locator Customization: by Paul Taylor – 1.01
WooCommerce Sequential Order Numbers Pro: by SkyVerge – 1.11.3 – Installed version not tested with active version of WooCommerce 4.4.1
WooCommerce Xero Integration: by WooCommerce – 1.7.33 – Installed version not tested with active version of WooCommerce 4.4.1
WP Downgrade | Specific Core Version: by Reisetiger – 1.2.2
WP Store Locator: by Tijmen Smit – 2.2.233
Xero Sequential Numbers Addon: by Paul Taylor – 1.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #9 - /shop/
Cart: #10 - /cart/
Checkout: #11 - /checkout/
My account: #12 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty
Version: 1.5
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Memberships ###

Restriction mode:: Hide content only
Public content:: –
Excluded from discounts:: –
Membership plans:: 2

### Action Scheduler ###

Complete: 63
Oldest: 2020-08-27 13:24:26 +0000
Newest: 2020-09-24 09:21:03 +0000


```
</details>

**Comment by mrgrt** [comment]
Also experiencing this issue.

**Comment by ldr2273** [comment]
This has come up again in 3407639-zen.

Just notating that I was able to successfully reproduce this on a default install:

Here's a manual order I created with the order status moving from `On-hold` to `Pending` to `Completed`:

![Order](https://d.pr/i/bai4I2+)
Link: https://d.pr/i/bai4I2

And the log shows that the New Order email was sent twice: 

![WP Mail logging](https://d.pr/i/ZopNpl+)
Link: https://d.pr/i/ZopNpl

CC @tammullen- in case you are available for triaging (and since it's come up a couple of times now). TIA!

**Comment by juliaamosova** [comment]
Hi @beesmart,

Apologies for the delayed response! 

Thank you for taking the time to report this bug, we really appreciate your help. I can reproduce it on my end. 

Documenting the steps I took:

1. Placed order in `On hold` status - received new order notification email as Admin
2. Changed order status from `On hold` to `Pending payment` - no email 
3. Changed order status from `Pending payment` to `Completed` - received identical new order notification email as Admin in step 1.

![emails](https://user-images.githubusercontent.com/19143190/103318988-3387f100-49fe-11eb-9d95-578f77dc1993.jpg)

I am giving this issue a `high` priority as it is related to one of our critical flows of receiving emails as Admin of the site. 

We’re going to solve this bug as soon as possible.
